model/src/main/resources/META-INF/openapi.yaml

```diff
@@ -189,3 +189,50 @@ components:
           - type: BRANCH
             hash: "2e1cfa82b035c26cbbbdae632cea070514eb8b773f616aaeaf668e2f0be8f10d"
             name: "main"
+    referencesResponseWithMetadata:
+      value:
+        token: null
+        hasMore: false
+        references:
+          - type: BRANCH
+            hash: "84c57a20c5e956af4af40f3cc34ecc8a9028b9586da135c79011b1867aa3191d"
+            name: "main"
+            metadataProperties:
+              headCommitMeta:
+                hash: "84c57a20c5e956af4af40f3cc34ecc8a9028b9586da135c79011b1867aa3191d"
+                committer: ""
+                author: "nessie-author"
+                signedOffBy: null
+                message: "update table"
+                commitTime: "2021-11-26T08:01:13.855974Z"
+                authorTime: "2021-11-26T08:01:13.852826Z"
+                properties: {}
+          - type: BRANCH
+            hash: "da086850076827d2989c8ee1d7fd22f152f525d46a0441f2b22ad8119c0bbbe5"
+            name: "dev"
+            metadataProperties:
+              numCommitsAhead: 1
+              numCommitsBehind: 2
+              commonAncestorHash: "2e1cfa82b035c26cbbbdae632cea070514eb8b773f616aaeaf668e2f0be8f10d"
+              headCommitMeta:
+                hash: "da086850076827d2989c8ee1d7fd22f152f525d46a0441f2b22ad8119c0bbbe5"
+                committer: ""
+                author: "nessie-author"
```

Comment:
This will deserialize to a plain `Map` not to `CommitMeta`, right? Would it not be nice to expose the same value type in this case as in `LogResponse`?

Reply:
That's true. Maybe just move the commit-meta to a separate field. The other attribtues are simple/primitive ones.

Reply:
let me think about it a bit more how an implementation would actually look like. We agreed with @dimas-b to look at this in a follow-up